src/ClientRuntime/Microsoft.Rest.ClientRuntime/ServiceClient.cs

```diff
@@ -360,16 +387,23 @@ protected void InitializeHttpClient(HttpClientHandler httpClientHandler, params
                     {
                         handler = handler.InnerHandler as DelegatingHandler;
                     }
+
                     handler.InnerHandler = currentHandler;
                     currentHandler = handlers[i];
                 }
             }
 
-            var newClient = new HttpClient(currentHandler, false);
+            if (httpClient == null)
+            {
+                HttpClient = new HttpClient(currentHandler, false);
+            }
+            else
+            {
+                HttpClient = httpClient;
+            }
+
             FirstMessageHandler = currentHandler;
-            HttpClient = newClient;
-            Type type = this.GetType();
-            SetUserAgent(type.FullName, ClientVersion);
+            SetUserAgent(this.GetType().FullName, ClientVersion);
```

Comment:
We need to check if we can set DefaultRequestHeaders on an http client that has already been used for communication - if so, we should add our header values, without clearing the existing set of UserAgents